snapcraft/internal/repo/_base.py

```diff
@@ -217,6 +222,49 @@ def normalize(self, unpackdir):
         self._fix_xml_tools(unpackdir)
         self._fix_shebangs(unpackdir)
 
+    def read_source_package_attr(self, path: str) -> Optional[str]:
+        if sys.platform != "linux":
+            raise RuntimeError("xattr support only available for Linux")
+
+        # Extended attributes do not apply to symlinks.
+        if not os.path.islink(path):
+            return None
+        try:
+            source = xattr.getxattr(path, "user.snapcraft.source_package")
+        except OSError:
+            # OSError is raised if no xattr found.
+            return None
+
+        source = source.decode().strip()
+        if source:
+            return source
+        return None
+
+    def write_source_package_attr(self, path: str, source: str) -> None:
+        if sys.platform != "linux":
+            raise RuntimeError("xattr support only available for Linux")
+
+        # Extended attributes do not apply to symlinks.
+        if not os.path.islink(path):
+            xattr.setxattr(path, "user.snapcraft.source_package", source.encode())
+
+    def mark_source_package_contents(
```

Comment:
Is there a reason for this to be public? I would make it private.

Reply:
agreed